src/channel-details.js

```diff
@@ -0,0 +1,211 @@
+const collect = require('collect-stream')
+const { stableSort, merge } = require('./util')
+
+class ChannelDetailsBase {
+  constructor(channelName) {
+    this.name = channelName
+
+    this.members = new Set()
+    this.mentions = []
+    this.virtualMessages = []
+    this.newMessageCount = 0
+    this.datesSeen = new Set()
+    /* TODO: 
+    use cursor to remember scrollback state and fetch 
+    from leveldb.  
+    maybe store a negative offset to look up?*/
+    this.lastRead = 0 /* timestamp in epoch time */
+    this.joined = false
+    this.opened = false
+    this.topic = ''
+  }
+
+  toString() {
+    return this.name
+  }
+
+  addMember (key) {
+    this.members.add(key)
+  }
+
+  removeMember(key) {
+    this.members.delete(key)
+  }
+
+  getMembers () {
+    return Array.from(this.members)
+  }
+
+  addMention(mention) {
+    if (!this.opened) {
+      this.mentions.push(mention)
+    }
+  }
+
+  getMentions() {
+    return this.mentions.slice() // return copy
+  }
+
+  handleMessage(message) {
+    if (!this.opened) {
+      // ++var is an optimization:
+      // var++ creates a temporary variable while ++var doesn't
+      ++this.newMessageCount
+    }
+  }
+  
+  getNewMessageCount() {
+    return this.newMessageCount
+  }
+
+  markAsRead() {
+    this.lastRead = Date.now()
+    this.newMessageCount = 0
+    this.mentions = []
+  }
+
+  open() {
```

Comment:
This is more nitpicky, but would `focus`/`unfocus` be more descriptive do you think?